### PR TITLE
[feature] Add JSDocs to codebase

### LINKED ISSUE #50: [feature] Require JSDocs
### Is your feature request related to a problem? Please describe

JSDocs should be required https://jsdoc.app/about-getting-started.html

### Describe the solution you'd like

Add JSDocs as an eslint requirement and retroactively add in JSDocs to existing code

### Describe alternatives you've considered

N/A

### Additional context

N/A

### PR DESCRIPTION
**Pull request by andrewtdiz**
### Purpose of pull request

See title

**Related Issue**: resolves #50 

Allows for inline commenting to export into a JSDoc

### Pull request checklist

- [x] Branch and PR are named correctly
- [x] Updated relevant documentation
- [x] Tested with a tester bot
- [ ] Wrote unit tests for changes

### Testing instructions

Write testing instructions here

### CODE REVIEW

File: src/constants.js [hunk 1]
```diff
@@ -1,18 +1,28 @@
...
 };
 
-// Variables
+/**
+ * Define the prefix that all server commands must begin with
+ */
 export const prefix = '++';
+
+/**
+ * Define the vald commands that the bot will repsond to
```
Comment (brendacs) [inline src/constants.js:24]: respond - typo

File: src/utils/dispatcher.js [hunk 2]
```diff
@@ -1,5 +1,9 @@
 import client from '../client';
 
+/**
+ * Parse message into its component args and emits back to the client
+ * @param {Object} message = the original message object emitted by a user
```
Comment (brendacs) [inline src/utils/dispatcher.js:5]: Let's change all the equal signs to dashes, similar to your mongoose comment

File: src/index.js [hunk 3]
```diff
@@ -5,6 +5,10 @@ import './client';
 import './bots/pins';
 import './bots/ping';
 
+/**
+ * Connect to MongoDB ip/port from .env
+ * @param {string} MONGODB - The string the ip/port for the MongoDB connection.
```
Comment (brendacs) [inline src/index.js:10]: typo here, erase "the string" since we already say it's a string param

**Review by brendacs — CHANGES_REQUESTED** [review]
Mostly good! Just needs a few minor changes

**Review by brendacs — APPROVED** [review]
Looks good! Just fix the merge conflicts and you can merge. Also, a note that JSDoc requirements were not added to the .eslintrc so won't affect linting, but we can do that in a different PR

File: src/client.js [hunk 4]
```diff
@@ -31,12 +37,16 @@ client
...
   client.login(BOT_TOKEN);
 }
 
-// Add message listener
+/**
+ * Listen for message and send to dispatch when one is received
+ */
 client.on('message', (message) => dispatchCommand(message));
 
 export default client;
```
Comment (sirNikolai) [inline src/client.js:52]: This should have a docstring while everything above have comments. Docstrings are used for variables and functions (especially exports), while the internal logic doesn't need that.

Comment (brendacs) [reply src/client.js:53]: Good catch, I wasn't checking what each one was on

File: src/index.js [hunk 5]
```diff
@@ -7,6 +7,10 @@ import './bots/comp';
 import './bots/ping';
 import './bots/pins';
 
+/**
+ * Connect to MongoDB ip/port from .env
+ * @param {string} MONGODB - the ip/port for the MongoDB connection.
+ */
 mongoose
```
Comment (sirNikolai) [inline src/index.js:14]: Same as my other comment above. Since this isn't a declaration or export it isn't really a docstring.

**Review by sirNikolai — CHANGES_REQUESTED** [review]

**Pull request merged**